program/src/processor.rs

```diff
@@ -110,12 +110,6 @@ fn get_stake_checked(
 }
 
 fn check_governance_exists(program_id: &Pubkey, governance_info: &AccountInfo) -> ProgramResult {
-    // Ensure the provided governance address is the correct address derived from
-    // the program.
-    if !governance_info.key.eq(&get_governance_address(program_id)) {
-        return Err(PaladinGovernanceError::IncorrectGovernanceConfigAddress.into());
-    }
-
```

Comment:
This is one of those "groundwork" changes that otherwise make you go "what the?".